lib/mage/MageError.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = class MageError extends Error {
+	constructor(data) {
+		super(data.message || 'Mage Error');
+
+		this.code = data.code || 'server';
+		this.level = data.level || 'error';
```

Comment:
If the level is unrecognized, should we default it to `alert` or something, so that it can be fixed?

Throwing or otherwise seems unwise, since the user is trying to log something here, and we shouldn't get in the way of that. But especially because of that, it does make sense to detect it, fix it on the spot, and escalate it.

Reply:
Defaulting to alert but logging a warning about the discrepancy.